### PR TITLE
fixed wrong filename generation through stamp tool #757

### LINKED ISSUE #757: Documents generated through the stamp tool are named FILENAME_watermarked.pdf
Not a big deal, but maybe they could be named FILENAME_stamped.pdf to differentiate with the watermark tool ?

Also, it would be nice to be able to customize that last part? 
Or maybe to use the locale the user has chosen to adapt it?

### PR DESCRIPTION
**Pull request by mannam11**
- fixed the wrong filename generation through stamp tool when downloaded.
 - changed variable names from 'watermark' to 'stamp' in StampController file, so that there will be no confusion in the future between these two.

Closes #757 

## Checklist:

- [x] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## Contributor License Agreement

By submitting this pull request, I acknowledge and agree that my contributions will be included in Stirling-PDF and that they can be relicensed in the future under the MPL 2.0 (Mozilla Public License Version 2.0) license.

(This does not change the general open-source nature of Stirling-PDF, simply moving from one license to another license)

### CODE REVIEW

**Review by sbplat — APPROVED** [review]

**Comment by sbplat** [comment]
Please complete the checklist as well, thanks!

**Comment by mannam11** [comment]
@sbplat I have reviewed and completed the checklist

**Comment by Frooodle** [comment]
ahh some conflicts now sorry could you resolve

File: src/main/java/stirling/software/SPDF/controller/api/misc/StampController.java [hunk 1]
```diff
@@ -88,6 +88,45 @@ public ResponseEntity<byte[]> addStamp(@ModelAttribute AddStampRequest request)
         // Load the input PDF
         PDDocument document = Loader.loadPDF(pdfFile.getBytes());
 
+        for (PDPage page : document.getPages()) {
```
Comment (Frooodle) [inline src/main/java/stirling/software/SPDF/controller/api/misc/StampController.java]: Duplicated code due to code merge I think?

Comment (Frooodle) [reply src/main/java/stirling/software/SPDF/controller/api/misc/StampController.java]: Replaced by l132

**Comment by mannam11** [comment]
@Frooodle did some changes but its broke.

**Comment by Frooodle** [comment]
No problem, let me fix :)

**Review by Frooodle — APPROVED** [review]

**Comment by mannam11** [comment]
Thanks @Frooodle

**Pull request merged**